test/abstractarray.jl

```diff
@@ -1389,7 +1390,67 @@ end
         @test [v v;;; fill(v, 1, 2)] == fill(v, 1, 2, 2)
     end
 
-    @test_throws BoundsError hvncat(((1, 2), (3,)), false, zeros(Int, 0, 0, 0), 7, 8)
+    # 0-dimension behaviors
+    # exactly one argument, placed in an array
+    # if already an array, copy, with type conversion as necessary
+    @test_throws ArgumentError hvncat(0)
+    @test hvncat(0, 1) == fill(1)
```

Comment:
Do we really need to allow this? Passing `()` for 0-d makes sense, but this form doesn't to me, since there is no dimension 0.

Reply:
`hvncat(::Int, args...)` is just a specialization for `hvncat(::Tuple{Vararg{Int}}, true, args...)`, so it continues this pattern:
```
`hvncat(3, ...)` == `hvncat((1, 1, n), ...)`.
hvncat(2, ...) == hvncat((1, n), ...)
hvncat(1, ...) == hvncat((n,), ...)
hvncat(0, ...) == hvncat((), ....)
```
That does mean, though, I could actually consolidate the `_typed_hvncat` methods for the 0-d cases to all refer to `_typed_hvncat(::Type{T}, ::Val{0}, args...)`, or vice versa.

Reply:
Ok, I guess the way to understand it is that it refers to the number of dimensions of the block array?

Reply:
Yeah, and it's the special case where there is only one dimension involved: `[a ;;; b ;;; c] => hvncat(3, a, b, c)` or `[a ;;;] => hvncat(3, a)`, which just bumps up the dimensions if ndims(a) < 3.